DataBaseTutorial/ProductAdd.xaml.cs

```diff
@@ -18,21 +15,35 @@ namespace DataBaseTutorial
     public partial class ProductAdd : Window
     {
         private readonly MainWindow _parent;
-        public string ProductName { get; set; }
+        public ValidationResult results = new ValidationResult();
 
         public ProductAdd(MainWindow parent)
         {
             _parent = parent;
             InitializeComponent();
-            
         }
 
         private void ButtonSaveNewProduct_Click(object sender, RoutedEventArgs e)
         {
-            var product = new Product(BoxProductName.Text, 01, 0.5, true, "01/02/2020 07:32:17");
-            _parent._myProducts.Add(product);            
+            ProductValidator validator = new ProductValidator();
+            ValidationResult results = validator.Validate(new Product(BoxProductName.Text, Convert.ToInt32(BoxProductID.Text), Convert.ToDouble(BoxProductWeight.Text), ((bool)BoxProductAddled.IsChecked == true), BoxProductDate.Text));
+            if (!results.IsValid)
```

Comment:
You can jyst use ValidateAndThrow method of validator. Then you get rid of ValidationResult and this construction at all